packages/client/src/views/OfficeHome/outbox/Outbox.tsx

```diff
@@ -0,0 +1,402 @@
+          iconId = `registering${index}`
+          icon = () => <Spinner id={iconId} key={iconId} size={24} />
+          statusText = formatMessage(statusSendingForApproval)
+          break
+        }
+      // eslint-disable-next-line no-fallthrough
+      case SUBMISSION_STATUS.REGISTERING:
+        iconId = `registering${index}`
+        icon = () => <Spinner id={iconId} key={iconId} size={24} />
+        statusText = formatMessage(statusRegistering)
+        break
+      case SUBMISSION_STATUS.REQUESTING_CORRECTION:
+        iconId = `requestingCorrection${index}`
+        icon = () => <Spinner id={iconId} key={iconId} size={24} />
+        statusText = formatMessage(statusRequestingCorrection)
+        break
+      case SUBMISSION_STATUS.READY_TO_REJECT:
+        iconId = `waiting${index}`
+        icon = () => <StatusWaiting id={iconId} key={iconId} />
+        statusText = formatMessage(statusWaitingToReject)
+        break
+      case SUBMISSION_STATUS.REJECTING:
+        iconId = `rejecting${index}`
+        icon = () => <Spinner id={iconId} key={iconId} size={24} />
+        statusText = formatMessage(statusRejecting)
+        break
+      case SUBMISSION_STATUS.READY_TO_REINSTATE:
+        iconId = `waiting${index}`
+        icon = () => <Spinner id={iconId} key={iconId} size={24} />
+        statusText = formatMessage(statusWaitingToBeReinstated)
+        break
+      case SUBMISSION_STATUS.REINSTATING:
+        iconId = `reinstating${index}`
+        icon = () => <Spinner id={iconId} key={iconId} size={24} />
+        statusText = formatMessage(statusReinstating)
+        break
+      case SUBMISSION_STATUS.READY_TO_ARCHIVE:
+        iconId = `waiting${index}`
+        icon = () => <StatusWaiting id={iconId} key={iconId} />
+        statusText = formatMessage(statusWaitingToBeArchived)
+        break
+      case SUBMISSION_STATUS.ARCHIVING:
+        iconId = `registering${index}`
+        icon = () => <Spinner id={iconId} key={iconId} size={24} />
+        statusText = formatMessage(statusArchiving)
+        break
+      case SUBMISSION_STATUS.READY_TO_CERTIFY:
+        iconId = `waiting${index}`
+        icon = () => <StatusWaiting id={iconId} key={iconId} />
+        statusText = formatMessage(statusWaitingToCertify)
+        break
+      case SUBMISSION_STATUS.CERTIFYING:
+        iconId = `waiting${index}`
+        icon = () => <Spinner id={iconId} key={iconId} size={24} />
+        statusText = formatMessage(statusCertifying)
+        break
+      case SUBMISSION_STATUS.FAILED_NETWORK:
+        iconId = `failed${index}`
+        icon = () => <StatusWaiting id={iconId} key={iconId} />
+        statusText = formatMessage(waitingToRetry)
+        break
+      default:
+        // default act as  SUBMISSION_STATUS[SUBMISSION_STATUS.READY_TO_REGISTER]:
+        iconId = `waiting${index}`
+        icon = () => <StatusWaiting id={iconId} key={iconId} />
+        statusText = formatMessage(statusWaitingToRegister)
+    }
```

Comment:
@tahmidrahman-dsi Could you please refactor this part? Probably using a map would be easier to manage then this switch statement

Reply:
Thanks very much, @Zangetsu101, will do.